src/libraries/Microsoft.PowerFx.Interpreter/Functions/Library.cs

```diff
@@ -2255,10 +2255,111 @@ public static FormulaValue Error(IRContext irContext, FormulaValue[] args)
                         return CommonErrors.RuntimeTypeMismatch(irContext);
                 }
 
-                result.Add(new ExpressionError { Kind = errorKind, Message = messageField?.Value as string });
+                var message = messageField != null ? messageField.Value : GetDefaultErrorMessage(errorKind);
+                result.Add(new ExpressionError { Kind = errorKind, Message = message });
             }
 
             return result;
+        }
+
+        private static string GetDefaultErrorMessage(ErrorKind errorKind)

```

Comment:
>GetDefaultErrorMessage [](http://example.com/codeflow?start=30&length=22)

Where are these coming from?  (PA?)

If somebody adds to ErrorKind, how do they know to update this? 
Can we add a test that just uses refleection to get each value of ErrorKind, calls this method, and ensures valid (eg, doesn't hit default case at end)

Reply:
Yes, the messages come from Power Apps. Good idea about the test, added one.